GSA_Adapter/Convert/ToGsa/Analysis/AnalysisStage.cs

```diff
@@ -0,0 +1,93 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Adapters.GSA;
+using BH.oM.Adapters.GSA.Analysis;
+using BH.oM.Base;
+using BH.oM.Structure.Constraints;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SurfaceProperties;
+using System;
+using System.Collections.Generic;
+
+namespace BH.Adapter.GSA
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public  Methods                           ****/
+        /***************************************************/
+
+        private static string ToGsaString(this AnalysisStage stage, string index)
+        {
+            string command = "ANAL_STAGE";
+            string name = stage.Name;
+
+            List<Object> elements = stage.Elements;
+            string objectIds = "";
+
+            for (int i = 0; i < elements.Count; i++)
+            {
+                string id = "";
+                int idInt; 
+
+                // String type
+                if (int.TryParse(elements[i].ToString(), out idInt))
+                {
+                    id = idInt.ToString();
+                }
+
+                // Property Type
+                else if (elements[i] is ISectionProperty)
+                {
+                    id = "PB" + (elements[i] as ISectionProperty).GSAId().ToString();
+                }
+                else if (elements[i] is LinkConstraint)
+                {
+                    id = "PL" + (elements[i] as LinkConstraint).GSAId().ToString();
+                }
+                else if (elements[i] is ISurfaceProperty)
+                {
+                    id = "PA" + (elements[i] as ISurfaceProperty).GSAId().ToString();
+                }
+
+                // Element type
+                else if (elements[i] is IBHoMObject) 
+                {
+                    id = (elements[i] as IBHoMObject).GSAId().ToString();
+                }
+                else
+                {
+                    BH.Engine.Reflection.Compute.RecordError("Unable to extract GSA Id. Supported types are doubles, ints, strings and BHoMObjects.");
+                    continue;
+                }
+
+                objectIds += " " + id;
+            }
+
```

Comment:
Some small tweaks here so the number property makes its way into GSA:

1. Would make the index be the number of the object, something like:
`index = stage.ToString();`
2.  After this is done, need to update the GSAId fragment on the stage, to make sure it matches the number. Something like this should help with that:
`stage.SetAdapterId(typeof(BH.oM.Adapters.GSA.GSAId), stage.Number);`
For this 2nd step to work, you will need to add a using: `using BH.Engine.Adapter;`